File: benchmarks/go/concurrent.go
package main

import (
	"fmt"
	"sync"
)

func worker(ch chan<- int64, wg *sync.WaitGroup) {
	defer wg.Done()
	var sum int64
	for i := int64(1); i <= 1000000; i++ {
		sum += i
	}
	ch <- sum
}

func main() {
	results := make(chan int64, 4)
	var wg sync.WaitGroup
	for i := 0; i < 4; i++ {
		wg.Add(1)
		go worker(results, &wg)
	}
	wg.Wait()
	close(results)
	var total int64
	for v := range results {
		total += v
	}
	fmt.Println(total)
}

File: benchmarks/python/concurrent.py
import threading

def worker(results, idx):
    s = sum(range(1, 1000001))
    results[idx] = s

results = [0] * 4
threads = [threading.Thread(target=worker, args=(results, i)) for i in range(4)]
for t in threads:
    t.start()
for t in threads:
    t.join()
print(sum(results))
